retrymanager_consistency.go

```diff
@@ -0,0 +1,122 @@
+package gocbcorex
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	"github.com/couchbase/gocbcorex/cbqueryx"
+	"github.com/couchbase/gocbcorex/memdx"
+)
+
+type RetryManagerConsistency struct {
+	Base              RetryManager
+	BucketChecker     BucketChecker
+	CollectionChecker CollectionChecker
+}
+
+func (m *RetryManagerConsistency) NewRetryController() RetryController {
+	return &retryControllerConsistency{
+		parent: m,
+		base:   m.Base.NewRetryController(),
+	}
+}
+
+type retryControllerConsistency struct {
+	parent *RetryManagerConsistency
+	base   RetryController
+}
+
+func (rc *retryControllerConsistency) ShouldRetry(ctx context.Context, err error) (time.Duration, bool, error) {
+	if errors.Is(err, memdx.ErrUnknownScopeName) {
+		if memdResErr, ok := err.(*memdx.ResourceError); ok {
+			if bucketErr, ok := err.(*KvBucketError); ok {
+				exists, existsErr := rc.parent.CollectionChecker.HasScope(ctx,
+					bucketErr.BucketName,
+					memdResErr.ScopeName)
+				if existsErr != nil {
+					return 0, false, &contextualError{
+						Message: "failed to double-check memd whether scope exists",
+						Cause:   existsErr,
+					}
+				}
+
+				if exists {
+					return 500 * time.Millisecond, true, nil
```

Comment:
Are these meant to always be a fixed 500ms?

Reply:
It is, since the timeline for these cases happening doesn't really benefit from exponential decay.